### ISSUE TITLE
Question about the dist embedding shape [64000, 6]

### ISSUE THREAD

**Issue by chris1220313648** [baaivision/URSA#6]
Hi, thanks a lot for releasing this great work!

I have a question about the embedding parameter used for dist.
In the code it seems that the embedding has shape [64000, 6].

How is this embedding obtained or computed?

Why is the dimension 6? My understanding is that the token embeddings of the base model are 4096-dimensional, so I’m wondering why a 6-dimensional embedding is used here instead.

Thanks in advance!

**Comment by PhyscalX** [comment]
For simplicity, we:

- Resize the token embeddings.
```python
qwen3_model.resize_token_embeddings(215669, mean_resizing=False)
```
- Remove Qwen3's ``lm_head`` and initialize a new 64K ``lm_head`` to reduce logits computation.
